concrete/src/Package/Package.php

```diff
@@ -733,219 +737,70 @@ public function upgrade()
-                $simpleAnnotationDriver = new \Doctrine\ORM\Mapping\Driver\AnnotationDriver($cachedSimpleAnnotationReader, $this->getPackageMetadataPaths());
-                return $simpleAnnotationDriver;
-            }else{
-                // Use default AnnotationReader
-                $cachedAnnotationReader = $this->app->make('orm/cachedAnnotationReader');
-                $annotationDriver = new \Doctrine\ORM\Mapping\Driver\AnnotationDriver($cachedAnnotationReader, $this->getPackageMetadataPaths());
-                return $annotationDriver;
-            }
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_XML){
-            $driverImpl = new \Doctrine\ORM\Mapping\Driver\XmlDriver($this->getPackageMetadataPaths());
-
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_YAML){
-            $driverImpl = new \Doctrine\ORM\Mapping\Driver\YamlDriver($this->getPackageMetadataPaths());
-        }
-        return $driverImpl;
-    }
-    
-    /**
-     * Get path to the location containing the metadata info
-     * 
-     * @return array 
-     */
-    public function getPackageMetadataPaths()
-    {
-        // annotations entity path
-        $paths = array();
-        if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_ANNOTATION){
-            // Support for the legacy method for backwards compatibility
-            $paths = $this->getPackageEntityPaths();
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_XML){
-            // return xml metadata dir
-            $paths =  array($this->getPackagePath() . DIRECTORY_SEPARATOR . REL_DIR_METADATA_XML);
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_YAML){
-            // return yaml metadata dir
-            $paths =  array($this->getPackagePath() . DIRECTORY_SEPARATOR . REL_DIR_METADATA_YAML);
-        }
-
-        // Add additional source path to the default namespace
-        // if pkgAutoloaderMapCoreExtensions is true
-        $corePath = !$this->enableLegacyNamespace() ? $this->getPackagePath() . DIRECTORY_SEPARATOR . DIRNAME_CLASSES . DIRECTORY_SEPARATOR . 'Concrete' : '';
-        if(!empty($corePath)){
-            $paths[] = $corePath;
-        }
-
-        $returnPaths = array();
-        foreach($paths as $path) {
-            if (is_dir($path)) {
-                $returnPaths[] = $path;
-            }
+        $em = $this->getPackageEntityManager();
+        if (is_object($em)) {
+            $this->destroyProxyClasses($em);
+            $this->installEntitiesDatabase();
         }
-
-        return $returnPaths;
+        static::installDB($this->getPackagePath() . '/' . FILENAME_PACKAGE_DB);
     }
-    
-    /**
-     * Get relative paths to the location containing the metadata info
-     * 
-     * @return array 
-     */
-    public function getPackageMetadataRelativePaths()
```

Comment:
Since this is a part of the public API, we should deprecate it and the other methods that were removed instead of completely removing them.